### ISSUE TITLE
Add more logs options and add logs in diary

### ISSUE THREAD

**Issue by crustamet** [AamuLumi/Rimworld-Diary#1]
I always wanted to write stories but this game writes the story for you.
Can we please import all chats that pawns had everyday this would be such a good addition to this diary mod.

**Comment by AamuLumi** [comment]
Hey,

Do you want a display for pawss chats like the one existing for events in the mod or do you want the chats to be imported directly in the diary ?
Personally, I think the first idea is better because in a medium / large colony, there's too much information directly entered in the diary and it will become something like a logger instead of a real diary.

**Comment by crustamet** [comment]
I think better should be an option because two things. i know this is just a game and the social interractions between pawns and all that jazz are just logs instead of diary but after my story is finished i can polished it after.

The dates and times must be accurate so if the event happens if two pawns interracted i should have seen that after the event..

Anyway I can comment after implemented :D if i'm not being annoying ^^

**Comment by crustamet** [comment]
![image](https://user-images.githubusercontent.com/36475250/202450957-279d2186-8d07-4729-ac61-484e9f6a2815.png)

**Comment by crustamet** [comment]
Another Idea is to make another option to write all actions of all pawns into the diary ^^ i know it will be a mess but the story is the story bro. i would love to see something like this in the game, i can share the story afterwards with my pals.

Anyway im not playing the game with 100 characters im only playing with 2 or 3 always keep dying when i have like 6 - 10 pawns, and im always playing at the hardest difficulty just to "read a new story like" of what could have happend if

**Comment by crustamet** [comment]
like these guys have a plague
for sure they social interract with each other

![image](https://user-images.githubusercontent.com/36475250/202452610-d7a600c3-f81f-4adf-909b-dc28d01dc1a0.png)

**Comment by crustamet** [comment]
Also these actions should be written in the diary also.

![image](https://user-images.githubusercontent.com/36475250/202454043-850bb780-aa12-4213-a5e8-93abed5aa3c3.png)

**Comment by AamuLumi** [comment]
Okay I think I understand what you want.

Here's the features : 
- add more informations in the Diary logs displayed, with possibility of filtering informations. Filters are : "All", "Events", "Chats" and "Actions".
- add a setting to add the logs informations in the diary. The setting can be "None", "Events", "Chats", "Actions", "All".

In the diary, informations will be written like this : 
```
[8h] Page : Reducing resistance of McTodd
[7h] Page encouraged McTodd to discuss goodwill
```

Does it seem correct to you ?

I don't know if pawns actions are logged somewhere in the game and how to write actions in the diary at the moment they occur. I'll check that when I'll work again on the mod.

**Comment by crustamet** [comment]
Yes this is correct bro :D 
Im sure actions are logged :D somewhere.. Don't know the programming stuff.
Hope you find it fast without too much fuss when you have the time.

**Comment by AamuLumi** [comment]
Hey,

I worked today on the feature, and I got this (with write filter setting set to "All") : 

![image](https://user-images.githubusercontent.com/1839713/202860844-a372d08e-fb92-4faf-a862-cb6ecf497d42.png)

I remarked "Reducing resistance" events conducts to social events. Another thing : all actions are not logged somewhere in Rimworld.
For these 2 reasons, I only developed the possibility to log and add to diary social interactions, and not all actions of the game.

Does it seem enough for your case ?

**Comment by crustamet** [comment]
Surely its progress looks very nice. The thing is when I read the diary, I can imagine what happened in the real world if this is a story that is the point.
for now until Rimworld gets a better social interactions or some mod that does that very detailed stories that connects with events this is very good. 
Another thing the update for the mod is out ? or ?

**Comment by AamuLumi** [comment]
I'm currently doing some small final fixes for this feature and the #2.
I expect to release it today.

**Comment by AamuLumi** [comment]
The feature is now updated on Steam.

If you find some problem relative to the discussed feature, you can reopen this issue or create a new issue.

Love on you, and long life to your pawns.
